modules/oe_newsroom_newsletter/src/Form/UnsubscribeForm.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_newsroom_newsletter\Form;
+
+use Drupal\Component\Utility\Html;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\oe_newsroom\Exception\InvalidApiConfiguration;
+use Drupal\oe_newsroom_newsletter\Api\NewsroomClient;
+use Drupal\oe_newsroom_newsletter\Api\NewsroomClientInterface;
+use GuzzleHttp\Exception\BadResponseException;
+use GuzzleHttp\Exception\ServerException;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Subscribe Form.
+ *
+ * Arguments:
+ *  - A distribution lists array
+ *    ex. array(0 => array('sv_id' => 20, 'name' => 'XY Newsletter'))
+ */
+class UnsubscribeForm extends NewsletterFormBase {
+
+  /**
+   * API for newsroom calls.
+   *
+   * @var \Drupal\oe_newsroom_newsletter\Api\NewsroomClientInterface
+   */
+  protected $newsroomClient;
+
+  /**
+   * {@inheritDoc}
+   */
+  public function __construct(NewsroomClientInterface $newsroomClient) {
+    $this->newsroomClient = $newsroomClient;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      NewsroomClient::create($container)
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'oe_newsroom_newsletter_unsubscribe_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, array $distribution_lists = []) {
+    $currentUser = $this->currentUser();
+
+    $form['#id'] = Html::getUniqueId($this->getFormId());
+
+    // Start building the form itself.
+    $form['email'] = [
+      '#type' => 'email',
+      '#title' => $this->t('Your e-mail'),
+      '#default_value' => $currentUser->isAnonymous() ? '' : $currentUser->getEmail(),
+      '#required' => TRUE,
+    ];
+    if (count($distribution_lists) > 1) {
+      $options = array_column($distribution_lists, 'name', 'sv_id');
```

Comment:
Since this code is the same in the two blocks, why not use the form base class to return these parts?.